tests/unit/core/utils/test_url.js

```diff
@@ -0,0 +1,80 @@
+import url from 'url';
+
+import { addQueryParams, removeUndefinedProps } from 'core/utils/url';
+
+describe(__filename, () => {
+  describe('removeUndefinedProps', () => {
+    it('removes undefined properties', () => {
+      expect(removeUndefinedProps({ thing: undefined })).toEqual({});
+    });
+
+    it('preserves falsy properties', () => {
+      expect(removeUndefinedProps({ thing: false })).toEqual({ thing: false });
+    });
+
+    it('preserves other properties', () => {
+      expect(removeUndefinedProps({ thing: 'thing' })).toEqual({
+        thing: 'thing',
+      });
+    });
+
+    it('does not modify the original object', () => {
+      const example = { thing: undefined };
+      removeUndefinedProps(example);
+      expect(example).toEqual({ thing: undefined });
+    });
+  });
+
+  describe('addQueryParams', () => {
+    it('adds a query param to a plain url', () => {
+      const output = addQueryParams('http://whatever.com/', { foo: 'bar' });
+      expect(url.parse(output, true).query).toEqual({ foo: 'bar' });
+    });
+
+    it('adds more than one query param to a plain url', () => {
+      const output = addQueryParams('http://whatever.com/', {
+        foo: 'bar',
+        test: 1,
+      });
+      expect(url.parse(output, true).query).toEqual({ foo: 'bar', test: '1' });
+    });
+
+    it('overrides an existing parameter', () => {
+      const output = addQueryParams('http://whatever.com/?foo=1', {
+        foo: 'bar',
+      });
+      expect(url.parse(output, true).query).toEqual({ foo: 'bar' });
+    });
+
+    it('overrides multiple existing parameters', () => {
+      const output = addQueryParams('http://whatever.com/?foo=1&bar=2', {
+        foo: 'bar',
+        bar: 'baz',
+      });
+      expect(url.parse(output, true).query).toEqual({ foo: 'bar', bar: 'baz' });
+    });
+
+    it('leaves other params intact', () => {
+      const output = addQueryParams('http://whatever.com/?foo=1&bar=2', {
+        bar: 'updated',
+      });
+      expect(url.parse(output, true).query).toEqual({
+        foo: '1',
+        bar: 'updated',
+      });
+    });
+
+    it('handles relative URLs', () => {
+      const output = addQueryParams('/relative/path/?one=1', { two: '2' });
+      expect(output).toMatch(/^\/relative\/path\//);
+      expect(url.parse(output, true).query).toEqual({ one: '1', two: '2' });
+    });
+
+    it('removes undefined query parameters', () => {
+      const output = addQueryParams('http://whatever.com/', {
+        bar: undefined,
+      });
+      expect(url.parse(output, true).query).toEqual({});
+    });
+  });
+});
```

Comment:
can you point to where `.src` is sanitized?

Reply:
I don't think we have any special. What's your concern?

Reply:
That `src="><script src=\"....` or something less scary but still breaky.

Reply:
That is already possible: https://addons.mozilla.org/en-US/firefox/addon/tranquility-1/?src=src=%22%3E%3Cscript%20src=\%22.... but it is treated as a plain string and react does auto-escaping so we do not do anything special.